components/style/stylist.rs

```diff
@@ -506,11 +545,15 @@ impl Stylist {
                 }
                 #[cfg(feature = "gecko")]
                 CssRule::FontFace(ref rule) => {
-                    _extra_data.add_font_face(&rule, origin);
+                    _extra_data
+                        .borrow_mut_for_origin(&origin)
```

Comment:
This can move to the outer loop I guess, not sure if worth, given you'd need to `cfg` it somehow.